teuthology/task/kernel.py

```diff
@@ -188,10 +188,21 @@ def need_to_install(ctx, role, version):
                                                                  want=version))
 
     if '.' in str(version):
-        # version is utsrelease, yay
         if cur_version == version:
             log.debug('utsrelease strings match, do not need to install')
             ret = False
+        our_os = teuthology.get_distro(ctx)
+        log.debug("Distro of this test job: {}".format(our_os))
+        if our_os in ['sle', 'opensuse']:
+            cur_version_match = re.search('(.*)-default$', cur_version)
+            if cur_version_match:
+                cur_version_rp = cur_version_match.group(1)
+                if cur_version_rp in version:
+                    log.debug('"{}" is a substring of "{}" - latest opensuse kernel is running'
+                              .format(cur_version_rp, version))
+                    ret = False
+            else:
+                log.debug('failed to parse current kernel version (opensuse codepath)')
```

Comment:
replace `(opensuse codepath)` with output of `get_distro()`?

Reply:
also maybe provide `cur_version` value to the user to make it clear what cannot be parsed for debugging purpose

Reply:
done